module2/Homework2-start/ColorPicker/ContentView.swift

```diff
@@ -0,0 +1,167 @@
+import SwiftUI
+
+struct ContentView: View {
+  @Environment(\.verticalSizeClass) var verticalSizeClass
+  @Environment(\.horizontalSizeClass) var horizontalSizeClass
+
+  @State private var alertIsVisible: Bool = false
+  @State private var redColor: Double = Constants.defaultRedColor
+  @State private var greenColor: Double = Constants.defaultGreenColor
+  @State private var blueColor: Double = Constants.defaultBlueColor
+
+  // When the app first runs, the default for previewColor shows up as white and not these values. Why?
+  @State private var previewColor: Color =  Color(red: Constants.defaultRedColor, green: Constants.defaultGreenColor, blue: Constants.defaultBlueColor)
+  
+  var body: some View {
+    
+    ZStack {
+      // The size class vars are not getting set and evaluating this if-statement. Why?
+      if verticalSizeClass == .compact && horizontalSizeClass == .regular {
+        // Handle landscape
+        HStack(spacing: 20) {
+          ColorPreviewView(previewColor: $previewColor)
+          ColorSlidersView(redColor: $redColor, greenColor: $greenColor, blueColor: $blueColor, previewColor: $previewColor)
+        }
+      } else {
+        // Default to portriat layout
+        VStack(spacing: 20) {
+          ColorPreviewView(previewColor: $previewColor)
```

Comment:
Don't use binding here. There's no reason to use binding unless you want there to be two-way communication between the functions.

Reply:
Good catch!